grtcirc.cc

```diff
@@ -68,10 +58,15 @@ double radtometers(double rads)
   return (rads * EARTH_RAD);
 }
 
-double gcdist(double lat1, double lon1, double lat2, double lon2)
+double gcdist(PositionRad pos1, PositionRad pos2)
 {
   errno = 0;
 
+  const double lat1 = pos1.latR;
+  const double lon1 = pos1.lonR;
+  const double lat2 = pos2.latR;
+  const double lon2 = pos2.lonR;
+
```

Comment:
I agree it's weird.  Evidently they are trying to avoid domain errors in asin, the clamps on zr, xr are harder to justify.

The clamp and EDOM code you cited came in at https://github.com/GPSBabel/gpsbabel/commit/644c09b5e "Check valid range for n before calling acos(n). Set errno always to zero."  Apr 29, 2006